pyinfra/facts/apt.py

```diff
@@ -94,3 +118,39 @@ def command(self) -> str:
 
     def requires_command(self) -> str:
         return "apt-key"
+
+
+class AptSimulationDict(TypedDict):
+    upgraded: int
+    newly_installed: int
+    removed: int
+    not_upgraded: int
+
+
+class SimulateOperationWillChange(FactBase[AptSimulationDict]):
+    """
+    Simulate an 'apt-get' operation and try to detect if any changes would be performed.
```

Comment:
Didn't mention in my review, this is a very cool trick, love it

Reply:
Technically speaking, this primarily works around the inability to tell pyinfra if a command run actually changed something.

Ideally I'd run the `apt-get` command and additionally inform pyinfra after it run if it actually changed something.

But glad you like it, and it does improve my update scripts a bit by avoiding unnecessary reboots :)